### PR TITLE
Whitelist $EZSQL_ERROR in variable naming conventions sniff

### PR DESCRIPTION
**Pull request by JDGrimes**
Though obscure, this global variable is used by WordPress core in `wpdb`, as a remnant of when it was forked from EzSQL. See https://core.trac.wordpress.org/browser/branches/4.6/src/wp-includes/wp-db.php#L1350.

See #691 for previous discussion.

### CODE REVIEW

**Comment by westonruter** [comment]
Tests are failing on PHPCS `master` and on PHP 5.2. 

In `master` it looks like the `phpcbf` output format has changed, and for PHP 5.2 it looks like its choking on a namespace separator.

**Comment by JDGrimes** [comment]
@westonruter See https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/696 as to why this is failing on `master`. `develop` is currently failing on `master` as well.

File: WordPress/Sniffs/NamingConventions/ValidVariableNameSniff.php [hunk 1]
```diff
@@ -57,6 +57,15 @@ class WordPress_Sniffs_NamingConventions_ValidVariableNameSniff extends PHP_Code
 	);
 
 	/**
+	 * Mixed-case variables used by WordPress.
+	 *
+	 * @var array
+	 */
+	public $wordpress_mixed_case_vars = array(
+		'EZSQL_ERROR',
```
Comment (jrfnl) [inline WordPress/Sniffs/NamingConventions/ValidVariableNameSniff.php]: 1. `public` properties within sniffs can be overruled by setting the property in the ruleset. I think the visibility here needs to change to either `protected` or `private`.
2. The pattern of `array( 'string_key' =>true )` for array properties would allow for checking using `isset()` rather than the slower `in_array()`.

Comment (JDGrimes) [inline WordPress/Sniffs/NamingConventions/ValidVariableNameSniff.php]: I was just following the pattern already set with the sniff's other similar properties. Obviously, using `protected` and `isset()` is better, though. Changing the other properties would break back-compat, so I haven't modified them.

Comment (jrfnl) [inline WordPress/Sniffs/NamingConventions/ValidVariableNameSniff.php]: As the `$php_reserved_vars` and `$whitelisted_mixed_case_member_var_names` properties look like they were never intended to be overloadable, I would have no problem breaking backward compatibility in this case.
If someone would currently be overloading them, they are effectively overwriting our defaults which would be a pretty bad idea.

Comment (JDGrimes) [inline WordPress/Sniffs/NamingConventions/ValidVariableNameSniff.php]: OK, but let's make a separate PR for that?

Comment (jrfnl) [inline WordPress/Sniffs/NamingConventions/ValidVariableNameSniff.php]: Fine by me ;-)

**Comment by jrfnl** [comment]
@JDGrimes Could you rebase (& squash) so that we can get the travis checks to pass ? I can't merge until they do :disappointed:
